src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserWithItems(user.getEmail())
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find shopping cart by user id: "
+                                + user.getId()));
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setTotal(countTotal(shoppingCart));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(dto.shippingAddress());
+        Set<OrderItem> orderItems = createOrderItems(order,shoppingCart.getCartItems());
+        order.setOrderItems(orderItems);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> findAllOrderItemsByOrderId(String email, Long orderId) {
+        Order order = getOrderByUserId(email, orderId);
+        return orderItemRepository.findAllByOrder(order)
+                .stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long orderId, StatusDto dto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId)
+        );
+        if (containsValidStatus(dto.status())) {
+            order.setStatus(dto.status());
+            return orderMapper.toDto(orderRepository.save(order));
+        } else {
+            throw new RuntimeException("Invalid status value");
+        }
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        orderRepository.deleteById(id);
+    }
+
+    private boolean containsValidStatus(Status status) {
+        return EnumSet.allOf(Status.class).contains(status);
+    }
+
+    private Set<OrderItem> createOrderItems(Order order, Set<CartItem> cartItems) {
+        Set<OrderItem> orderItems = new HashSet<>();
+        for (CartItem cartItem : cartItems) {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setOrder(order);
+            orderItems.add(orderItem);
+        }
+        return orderItems;
+    }
+
+    private BigDecimal countTotal(ShoppingCart shoppingCart) {
+        return BigDecimal.valueOf(shoppingCart.getCartItems()
+                .stream()
+                .mapToDouble(cartItem -> cartItem.getQuantity()
+                        * cartItem.getBook().getPrice().doubleValue())
+                .sum());
+    }
+
+    private User getUserByEmail(String email) {
+        return userRepository.findByEmail(email).orElseThrow(
+                () -> new EntityNotFoundException("Can't find user by email: " + email));
+    }
```

Comment:
I'd rather use `userService` method in case we have it, or move this one to there